src/runtime/x86_avx2.ll

```diff
@@ -1,3 +1,25 @@
+; Note that this is only used for LLVM 6.0+
+define weak_odr <32 x i8>  @pavgbx32(<32 x i8> %a, <32 x i8> %b) nounwind alwaysinline {
+  %1 = zext <32 x i8> %a to <32 x i32>
```

Comment:
Is it intentional to widen i8 to i32 here? (not i16?)

Reply:
Right, extending to 16-bit elements would be sufficient. I followed the SSE2 implementation for the sake of consistency, and FWIW the LLVM X86-backend tests do the same.
I can change the implementation to 16-bit extending if you think it should be done.

Reply:
I guess it doesn't matter too much here. If we ever tried to depend on LLVM's pattern matching directly, we would want to use 16 bits though.